Adding the navigation

marktrobinson opened the pull request:
Navigation for iOS only at the minute

mark-jTribe committed:
Adding the navigation

mark-jTribe committed:
Adding pop on complete

simonexmachina commented on index.ios.js:
```diff
@@ -6,9 +6,24 @@
+    View,
+    NavigatorIOS,
+    StyleSheet,
+  } = React;
```
Indentation

simonexmachina commented on src/components/app.js:
```diff
@@ -21,12 +21,19 @@ export default React.createClass({
   },
   render() {
-    //return <Circle user={this.state.user} />;
+    return <Circle user={this.state.user} onPress={ () => {
```
Maybe `onPress` would be better called `teamsPressed`?

simonexmachina commented on src/components/app.js:
```diff
@@ -21,12 +21,19 @@ export default React.createClass({
   render() {
-    //return <Circle user={this.state.user} />;
+    return <Circle user={this.state.user} onPress={ () => {
+        if (!this.currentUser.ref) return <View />;
```
This will just leave a blank screen, but the component should handle this internally anyway so leave it and I'll sort it out.

simonexmachina replied:
Actually, this does need to be handled here (you might want to show the teams for another user). So this should be:

```
this.currentUser.getRef().then(userRef => {
  this.props.navigator.push({
    ...
    passProps: {
      userRef,
      ...
```

simonexmachina commented on src/components/app.js:
```diff
@@ -21,12 +21,19 @@ export default React.createClass({
+            userRef: this.currentUser.ref,
+            onPress: this.goToTeam,
+          },
+        });
```
Also, given that this is handled inside the component in `Teams`, I reckon this should be moved into the `Circle` component, but I can do that if you like

simonexmachina commented on src/components/teams.js:
```diff
@@ -53,16 +56,29 @@ export default React.createClass({
+
+  onCompleteAddTeam(teamName) {
+    this.props.navigator.pop();
+  },
```
See how navigation concerns get spread throughout our code. This is another reason why routers are awesome - you can control your state transitions in a single, separate layer of your system.

simonexmachina commented:
Nice one, that's an :airplane: :+1:

simonexmachina committed:
Moved things around with some understanding of the Navigator

- Circle now loads its own data and handles its own navigation
- Teams handles its own navigation

simonexmachina commented:
Hey @mr16839072, FYI I added [a commit](https://github.com/jtribe/pomo/commit/30c20160b3ac29d0b22619e67029bc9c418e9062) which addressed my comments above.
